File: 9.回文数.cpp
/*
 * @lc app=leetcode.cn id=9 lang=cpp
 *
 * [9] 回文数
 */
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0)
            return false;
        std::string s = std::to_string(x);
        if(s.size() == 1)
            return true;
        int p1 = s.size()/2;
        int p2 = 0;
        if(s.size() & 1)
            p2 = s.size()/2;
        else
            p2 = s.size()/2-1;
        while(p1 < s.size() && p2 >= 0){
            if(s[p1--] != s[p2++])
                return false;
        }
        return true;
    }
};

File: 9.回文数.rs
/*
 * @lc app=leetcode.cn id=9 lang=rust
 *
 * [9] 回文数
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let x_str = x.to_string();
        let x_str_bytes = x_str.as_bytes();
        let mut head = 0;
        let mut tail = x_str.len() - 1;
        while head < tail {
            if x_str_bytes[head] != x_str_bytes[tail] {
                return false;
            }
            head += 1;
            tail -= 1;
        }
        return true;
    }
}
// @lc code=end

